stl/inc/xstring

```diff
@@ -2641,6 +2645,37 @@ private:
         }
     }
 
+#if _HAS_CXX20
+public:

```

Comment:
I worry about "going `public`" in a Standard-conditional block. In this case, it's immediately followed by `public:` on line 2679, so I think it would be simpler to simply unify these two and go `public:` before the `_HAS_CXX20` block. (It's simplest when a conditional block doesn't have to alter access control.)